analytic_partner_hr_timesheet_invoice/models/sale_order_line.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Tecnativa, S.L. - Pedro M. Baeza, Luis M. Ontalba
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.multi
+    def _compute_analytic(self, domain=None):
+        """Split by partner of the lines to filter only affected analytic
+        lines.
+        """
+        for partner in self.mapped('order_partner_id'):
+            so_lines = self.filtered(lambda x: x.order_partner_id == partner)
+            if not domain:
+                # To filter on analyic lines linked to an expense
+                expense_type_id = self.env.ref(
+                    'account.data_account_type_expenses',
+                    raise_if_not_found=False)
+                expense_type_id = expense_type_id and expense_type_id.id
```

Comment:
You have to recreate the domain in sale_timesheet, not this one (`domain = [('so_line', 'in', self.ids), '|', ('amount', '<=', 0.0), ('project_id', '!=', False)]`). You can even be more specific about the project if you prefer (and if it can be obtain - maybe through the analytic account -)